### PR TITLE
RMMIS-6422: Fix editing of previously saved input data

### PR DESCRIPTION
**Pull request by traviscooperscdhhs**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@scdhhs/ui-components",
-  "version": "1.12.0",
+  "version": "1.14.2",
```
Comment (mallenscdhhs) [inline package.json]: This doesnt need to change unless you are merging from master -> develop

File: src/Field.jsx [hunk 2]
```diff
@@ -174,9 +174,13 @@ export default React.createClass ({
    * @returns {JSX}
    */
   render() {
-    let fieldProps = (this.props.dependencyValue) ? (
-      Immutable.fromJS(this.props).set('disabled',this.state.disabled ? 'disabled' : false).toJSON()
-    ) : this.props;
+    let fieldProps = this.props;
+    if (this.props.dependencyValue) {
+      fieldProps = Immutable.fromJS(fieldProps).set('disabled', this.state.disabled ? 'disabled' : false).toJSON();
```
Comment (mallenscdhhs) [inline src/Field.jsx]: wouldnt you create a Map from `this.props`?

File: src/Field.jsx [hunk 3]
```diff
@@ -174,9 +174,13 @@ export default React.createClass ({
    * @returns {JSX}
    */
   render() {
-    let fieldProps = (this.props.dependencyValue) ? (
-      Immutable.fromJS(this.props).set('disabled',this.state.disabled ? 'disabled' : false).toJSON()
-    ) : this.props;
+    let immutableFieldProps = Immutable.fromJS(this.props);
```
Comment (mallenscdhhs) [inline src/Field.jsx]: I like the idea of naming Immutable variables so they convey what they are, however, I find `immutableFieldProps` to be a little too verbose. What if picked a convention that can convey that something is immutable but more terse than `immutable`? @kjschulz or @traviscooper, any suggestions? What about prefixing with `_`? `_fieldProps`?

File: src/Field.jsx [hunk 4]
```diff
@@ -174,9 +174,13 @@ export default React.createClass ({
...
+    if (this.props.dependencyValue) {
+      immutableFieldProps = immutableFieldProps.set('disabled', this.state.disabled ? 'disabled' : false);
+    }
+    if (this.state.value) {
```
Comment (mallenscdhhs) [inline src/Field.jsx]: what if `this.state.value` is falsy?

File: src/ValidationMixin.jsx [hunk 5]
```diff
@@ -8,7 +8,7 @@ module.exports = {
     Dispatcher.register( this.props.id + '-VALIDATE-FIELD' , function(action,data){
       if( action === constants.actions.FIELD_VALIDATION_ERROR &&
           data.id === this.props.id) {
-        this.setState({'hasError': data.hasError, 'errorMessage': data.errorMessage});
+        this.setState({'hasError': data.hasError, 'errorMessage': data.errorMessage, 'value': data.value});
```
Comment (mallenscdhhs) [inline src/ValidationMixin.jsx]: No need for single quotes around keys here

File: src/Field.jsx (hunk 3 above)
Comment (traviscooperscdhhs) [inline src/Field.jsx]: maybe prefix with `i` like `iFieldProps`.  I've always used `_` to denote internal prop of a class or entity.

File: src/Field.jsx (hunk 4 above)
Comment (traviscooperscdhhs) [inline src/Field.jsx]: sure, will check for depValues and stateValue

File: src/Field.jsx (hunk 3 above)
Comment (mallenscdhhs) [inline src/Field.jsx]: I like `i`, @kjschulz what say you?

**Pull request merged**